File: Python/underautomation/universal_robots/ssh/tools/common/ssh_connection_exception.py
import typing
from underautomation.universal_robots.ssh.tools.messages.transport.disconnect_reason import DisconnectReason
from underautomation.universal_robots.ssh.tools.common.ssh_exception import SshException
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Ssh.Tools.Common import SshConnectionException as ssh_connection_exception

class SshConnectionException(SshException):
	def __init__(self, message: str, disconnectReasonCode: DisconnectReason, inner: typing.Any, _internal = 0):
		if(_internal == 0):
			self._instance = ssh_connection_exception(message, disconnectReasonCode, inner)
		else:
			self._instance = _internal
	@property
	def disconnect_reason(self) -> DisconnectReason:
		return DisconnectReason(self._instance.DisconnectReason)

File: UnderAutomation.UniversalRobots.ObfuscatedSources/Ssh/Tools/Common/SshConnectionException.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Ssh.Tools.Messages.Transport;
using System.Runtime.Serialization;

namespace Ssh.Tools.Common {
	/// <summary>
	/// The exception that is thrown when connection was terminated.
	/// </summary>
	public class SshConnectionException : SshException, ISerializable {

		/// <summary>
		/// Initializes a new instance of the <xref href="UnderAutomation.UniversalRobots.Ssh.Tools.Common.SshConnectionException" data-throw-if-not-resolved="false"></xref> class.
		/// </summary>
		public SshConnectionException()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Initializes a new instance of the <xref href="UnderAutomation.UniversalRobots.Ssh.Tools.Common.SshConnectionException" data-throw-if-not-resolved="false"></xref> class.
		/// </summary>
		/// <param name="message">The message.</param>
		public SshConnectionException(string message)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Initializes a new instance of the <xref href="UnderAutomation.UniversalRobots.Ssh.Tools.Common.SshConnectionException" data-throw-if-not-resolved="false"></xref> class.
		/// </summary>
		/// <param name="message">The message.</param>
		/// <param name="disconnectReasonCode">The disconnect reason code.</param>
		public SshConnectionException(string message, DisconnectReason disconnectReasonCode)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Initializes a new instance of the <xref href="UnderAutomation.UniversalRobots.Ssh.Tools.Common.SshConnectionException" data-throw-if-not-resolved="false"></xref> class.
		/// </summary>
		/// <param name="message">The message.</param>
		/// <param name="disconnectReasonCode">The disconnect reason code.</param>
		/// <param name="inner">The inner.</param>
		public SshConnectionException(string message, DisconnectReason disconnectReasonCode, Exception inner)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Gets the disconnect reason if provided by the server or client. Otherwise None.
		/// </summary>
		public DisconnectReason DisconnectReason { get; }
	}
}
